core/app/[locale]/(default)/(faceted)/fetch-faceted-search.ts

```diff
@@ -158,10 +158,10 @@ type SearchProductsSortInput = Variables['sort'];
 type SearchProductsFiltersInput = Variables['filters'];
 
 interface ProductSearch {
-  limit?: number;
-  before?: string;
-  after?: string;
-  sort?: SearchProductsSortInput;
+  limit?: number | null;
+  before?: string | null;
+  after?: string | null;
+  sort?: SearchProductsSortInput | null;
```

Comment:
Couldn't we keep `undefined` and just pass `value ?? undefined` to prevent the extra `!== null` filter?

Reply:
The issue here is in the mismatch between what's parsed by `nuqs` and the Zod types. `nuqs` will use `null` for missing values. I personally dislike explicitly using `undefined` and prefer doing comparisons like `!= null`, etc.

Reply:
Gotcha.